actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -240,6 +260,12 @@ def signed_or_encrypted
         def upgrade_legacy_signed_cookies?
           request.secret_token.present? && request.secret_key_base.present?
         end
+
+        def upgrade_legacy_hmac_aes_cbc_cookies?
+          request.secret_key_base.present?                 &&
+            request.use_authenticated_cookie_encryption?   &&
+            request.use_authenticated_cookie_encryption?
```

Comment:
So this would be:

```ruby
request.secret_key_base.present? &&
  request.authenticated_encrypted_cookie_salt.present? &&
  request.encrypted_signed_cookie_salt.present? &&
  request.encrypted_cookie_salt.present?
```

And we can kill the `use_authenticated_cookie_encryption?` and `upgrade_legacy_hmac_aes_cbc_cookies?` methods then.

`authenticated_encrypted_cookie_salt` was placed up further, since it's less likely to be assigned on upgrading apps and we'll avoid the presence check on the present in older releases salt values.

Related: I'm not sure why we check the presence of the `secret_key_base` here. Rails apps won't boot without it these days.